p2p/service.py

```diff
@@ -142,6 +142,25 @@ def run_task(self, awaitable: Awaitable[Any]) -> None:
                 self.logger.trace("Task %s finished with no errors", awaitable)
         self._tasks.add(asyncio.ensure_future(_run_task_wrapper()))
 
+    def run_daemon_task(self, awaitable: Awaitable[Any]) -> None:
```

Comment:
I was looking at this and the `run_task` api and I think we should update them as follows.

```python
def run_task(self, fn, *args):
    @functools.wraps(fn)
    async def _run_task_wrapper():
        ...
    ...
```

By doing it this way we'll get intelligible stacktraces when tasks are not properly cleaned up which should make finding the offending code a lot easier.